src/controllers/v1/_common/demand.ts

```diff
@@ -38,6 +40,7 @@ export class DemandController extends Controller {
       logger,
       userUri: env.USER_URI,
     })
+    this.identity = container.resolve(Identity)
```

Comment:
is there a reason to not use `@injectable`?

Reply:
so I am trying it now I just pushed it. But when I use `@injectable` I have to pass it in the constructor like so: `
  constructor(private identity: Identity)` and then I din't know how to pass it e.g. in `v1/demandA`

Reply:
So if you pass it via `constructor(private identity: Identity)` the dependency injection framework should inject it for us. You won't have to instantiate it ourselves

Reply:
> That would make sense, but it doesn't do that - or at least not the way I made it ![Screenshot 2023-09-11 at 15 47 23](https://user-images.githubusercontent.com/58051865/267058412-2da3fcc7-204a-45dd-b320-929f250f974c.png)

Ok I see. This is because this is a helper class for the two demand controllers (under `demandA`/`demandB`). In this instance the `DemandController` class shouldn't be the `injectable` but the `DemandAController` and `DemandBController` should be. They can then pass the `identity` instance passed to them to the `DemandController`. Does that make sense?